### ISSUE TITLE
False-positive `else-if-used` with nested `if - else` block

### ISSUE THREAD

**Issue by cdce8p** [PyCQA/pylint#5340]
### Bug description

```py
import sys

if a:
    pass
elif b:
    pass
else:
    if sys.version_info >= (3, 8):   # false-positive: else-if-used
        pass
    else:
        pass
```

Notice the example above while working on https://github.com/PyCQA/astroid/pull/1258. Contrary to the suggestion, the `else - if` block can't easily be flattened here as there is another `else` block to consider.

The equivalent would be
```py
if a:
    ...
elif b:
    ...
elif sys.version_info >= (3, 8):
    pass
elif sys.version_info < (3, 8):
    pass
```
IMO this is more difficult to read the the original example.

### Configuration

```ini
[MASTER]
load-plugins=
    pylint.extensions.check_elif,
```

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
R5501: Consider using "elif" instead of "else if" (else-if-used)
```

### Expected behavior

No warning

### Pylint version

```shell
pylint 2.11.2-dev0
astroid 2.8.6-dev0
Python 3.10.0 (v3.10.0:b494f5935c, Oct  4 2021, 14:59:20) [Clang 12.0.5 (clang-1205.0.22.11)]
```

**Comment by Pierre-Sassoulas** [comment]
I don't think this is a false positive because this:

```python
sys_version = (3,7)
if a:
    ...
elif b:
    ...
elif sys_version >= (3, 8):
    print("3.8+")
elif sys_version < (3, 8):
    print("not 3.8+")
``` 

Is equivalent to this:

```python
sys_version = (3,7)
if a:
    ...
elif b:
    ...
elif sys_version >= (3, 8):
    print("3.8+")
else
    print("not 3.8+")
``` 

Because if ``sys.version_info >= (3, 8)`` is True, then ``sys.version_info < (3, 8)`` is False. (I replaced sys.version_info by a tuple so it's testable easily.)

**Comment by cdce8p** [comment]
You're right. I should have gone to bed earlier 😅
